src/analyses/local_safe_pointers.h

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************\
+
+Module: Local safe pointer analysis
+
+Author: Diffblue Ltd
+
+\*******************************************************************/
+
+/// \file
+/// Local safe pointer analysis
+
+#ifndef CPROVER_ANALYSES_LOCAL_SAFE_POINTERS_H
+#define CPROVER_ANALYSES_LOCAL_SAFE_POINTERS_H
+
+#include <goto-programs/goto_program.h>
+#include <util/std_expr.h>
+
+/// A very simple, cheap analysis to determine when dereference operations are
+/// trivially guarded by a check against a null pointer access.
+/// In the interests of a very cheap analysis we only search for very local
+/// guards -- at the moment only `if(x != null) { *x }`
+/// and `assume(x != null); *x` are handled. Control-flow convergence and
+/// possibly-alising operations are handled pessimistically.
+class local_safe_pointerst
+{
+  std::map<unsigned, std::set<exprt>> non_null_expressions;
+
+ public:
+  void operator()(const goto_programt &goto_program);
+
+  bool is_non_null_at_program_point(
+    const exprt &expr, goto_programt::const_targett program_point)
+  {
+    auto findit = non_null_expressions.find(program_point->location_number);
+    if(findit == non_null_expressions.end())
+      return false;
+    exprt tocheck = expr;
+    while(tocheck.id() == ID_typecast)
+      tocheck = tocheck.op0();
+    return findit->second.count(tocheck);
```

Comment:
Please use `findit->second.find(tocheck) != findit->second.end()` (or, if you insist, add `!= 0` to your code -- Visual Studio issues warnings for your code.